azurelinuxagent/common/protocol/wire.py

```diff
@@ -802,8 +829,17 @@ def check_wire_protocol_version(self):
     def upload_status_blob(self):
         ext_conf = self.get_ext_conf()
         if ext_conf.status_upload_blob is not None:
-            if not self.status_blob.upload(ext_conf.status_upload_blob):
-                self.get_host_plugin().put_vm_status(self.status_blob, ext_conf.status_upload_blob)
+            uploaded = False
+            try:
+                uploaded = self.status_blob.upload(ext_conf.status_upload_blob)
+            except (HttpError, ProtocolError) as e:
+                # errors have already been logged
```

Comment:
Even if we did that the actual upload would subsequently fail and still throw the protocol error.